### ISSUE TITLE
comment from hubzilla arriving as post with wrong attribution on friendica

### ISSUE THREAD

**Issue by git-marijus** [friendica/friendica#6347]
Here is the original comment:
https://pod.datamol.org/display/242697a244ab27eda02a45e19ab28f3e9c3f48db7fe2ddc7a4595a99805f7f22@pod.datamol.org

Here is what i see in friendica:
https://libranet.de/display/242697a244ab27eda02a45e19ab28f3e9c3f48db7fe2ddc7a4595a99805f7f22@pod.datamol.org

**Comment by AlfredSK** [comment]
Yep. It's known problem. We have investigated this some time ago (by looking at the raw XML data) and the conclusion was that these comments arrive as top level posts (the parent is missing) on Friendica's side.

Here is the discussion in our support forum (see comments). 
https://forum.friendi.ca/display/0b6b25a8-165b-fb74-e3a8-446991624943

Now that I read it again, it should be already fixed. But it turns out it isn't. :thinking:

**Comment by AlfredSK** [comment]
Here is what I have in my database:

Getting the URI of the item:
````
MariaDB [friendica]> select uri from item where guid = '242697a244ab27eda02a45e19ab28f3e9c3f48db7fe2ddc7a4595a99805f7f22@pod.datamol.org';
==>

https://hub.somaton.com/objects/242697a244ab27eda02a45e19ab28f3e9c3f48db7fe2ddc7a4595a99805f7f22@pod.datamol.org
````

Getting the source XML for that URI:
````
MariaDB [friendica]> select source from conversation where `item-uri` = 'https://hub.somaton.com/objects/242697a244ab27eda02a45e19ab28f3e9c3f48db7fe2ddc7a4595a99805f7f22@pod.datamol.org';
==>

<XML>
  <post>
    <status_message>
      <raw_message>\#Hubzilla for #Yunohost has been updated to version 3.8.8.  
  
[https://github.com/YunoHost-Apps/hubzilla\_ynh](https://github.com/YunoHost-Apps/hubzilla_ynh)</raw_message>
      <guid>242697a244ab27eda02a45e19ab28f3e9c3f48db7fe2ddc7a4595a99805f7f22@pod.datamol.org</guid>
      <diaspora_handle>mario@hub.somaton.com</diaspora_handle>
      
      <public>true</public>
      <created_at>2018-12-25 15:20:25 UTC</created_at>
      <provider_display_name>hubzilla</provider_display_name>
    </status_message>
  </post>
</XML>
````

**Comment by AlfredSK** [comment]
Maybe also noteworthy: the original parent post is a Hubzilla forum post.

**Comment by annando** [comment]
This is really nothing that we can do on our side, since Hubzilla is sending the wrong stuff.

**Comment by git-marijus** [comment]
I see... I did not have the logs... 
Closing

**Comment by redmatrix** [comment]
I'd like to request a review and perhaps re-open as you see fit. Yes, hubzilla was sending the wrong stuff, not through standard transmissions, but through the osolete and deprecated /p/xxxxxx.xml interface. We kept this open for as long as we supported the older redmatrix servers, but this ended on Jan. 1. This interface has now been removed, but is still present (and will still be present) on older servers. 

Friendica was requesting a copy of a *comment* by guid using this interface as part of the conversation completion mechanism. I believe this is wrong and the reason we answered with bad data. The purpose of this interface is to request a copy of the *parent* guid. If this had been done by Friendica, the bad data would not have been transferred and the old interface would not have been used. Presumably Friendica already had a signed copy of the comment and besides fetching it again, stored an unsigned and invalid representation of this object as a parent post instead of or in addition to the signed and valid comment it had already received.  

Anyway this is complicated and nobody is at fault, but both projects should try and remove this legacy code because a) you can get bad results from some servers if you ask for the wrong thing, and b) Friendica is apparently asking for the wrong thing. Let's just kill the deprecated interface on both sides and try to avoid any more recurrences of this or other related issues. Diaspora requested we do this a couple of years ago.

**Comment by annando** [comment]
We kept ```/p``` since at some time, Hubzilla hadn't offered the newer ```/fetch``` endpoint.

**Comment by AlfredSK** [comment]
Shouldn't we reopen this issue as a reminder?

Also, Hubzilla has removed the old-style /p/xxx.xml fetch interface today.
https://framagit.org/hubzilla/addons/commit/1d5c046e7ec58a31d91ab0c5d4d8ccfbad372d73

**Comment by MrPetovan** [comment]
Closed via #6392
